cpu/rpx0xx/periph/spi.c

```
@@ -0,0 +1,407 @@
+    return spi_config[spi].mosi_pin;
+}
+
+static void _poweron(spi_t spi)
+{
+    uint32_t reset_bit_mask =
+        (spi) ? RESETS_RESET_spi1_Msk : RESETS_RESET_spi0_Msk;
+
+    periph_reset(reset_bit_mask);
+    periph_reset_done(reset_bit_mask);
+}
+
+static void _poweroff(spi_t spi)
+{
+    uint32_t reset_bit_mask =
+        (spi) ? RESETS_RESET_spi1_Msk : RESETS_RESET_spi0_Msk;
+
+    periph_reset(reset_bit_mask);
+}
+
+void spi_init(spi_t spi)
+{
+    DEBUG("[rpx0xx] Call spi_init(spi=%" PRIdFAST8 ")", spi);
+
+    assert((unsigned)spi < SPI_NUMOF);
+
+    /* initialize device lock */
+    mutex_init(&locks[spi]);
+    /* trigger pin initialization */
+    spi_init_pins(spi);
+    /* clock prescaler values must be calculated */
+    pl022_clk[spi].clk = 0xff;
+}
+
+void spi_init_pins(spi_t spi)
+{
+    DEBUG("[rpx0xx] Call spi_init_pins(spi=%" PRIdFAST8 ")", spi);
+
+    assert((unsigned)spi < SPI_NUMOF);
+
+    const gpio_pad_ctrl_t mosi_pad_config = {
+        .drive_strength = DRIVE_STRENGTH_2MA,
+    };
+    const gpio_io_ctrl_t mosi_io_config = {
+        .function_select = FUNCTION_SELECT_SPI,
+    };
+    const gpio_pad_ctrl_t clk_pad_config = {
+        .drive_strength = DRIVE_STRENGTH_2MA,
+    };
+    const gpio_io_ctrl_t clk_io_config = {
+        .function_select = FUNCTION_SELECT_SPI,
+    };
+    const gpio_pad_ctrl_t miso_pad_config = {
+        .pull_up_enable = 1,
+        .input_enable = 1,
+        .schmitt_trig_enable = 1,
+    };
+    const gpio_io_ctrl_t miso_io_config = {
+        .function_select = FUNCTION_SELECT_SPI,
+    };
+
+    gpio_set_pad_config(spi_config[spi].mosi_pin, mosi_pad_config);
+    gpio_set_io_config(spi_config[spi].mosi_pin, mosi_io_config);
+    gpio_set_pad_config(spi_config[spi].clk_pin, clk_pad_config);
+    gpio_set_io_config(spi_config[spi].clk_pin, clk_io_config);
+    gpio_set_pad_config(spi_config[spi].miso_pin, miso_pad_config);
+    gpio_set_io_config(spi_config[spi].miso_pin, miso_io_config);
```

You could add a check `gpio_is_valid` for each pins.

---

Is fixed. I have put if-clauses around these calls. Would maybe an assert be better? Now it just fails silently.

---

Hmm I don't know if the assert is 100% suitable here. Maybe there are some use cases for MISO-only or MOSI-only ?